src/sagemaker/jumpstart/artifacts/model_uris.py

```diff
@@ -76,18 +94,35 @@ def _retrieve_model_uri(
         tolerate_deprecated_model=tolerate_deprecated_model,
     )
 
+    bucket_override = os.environ.get(ENV_VARIABLE_JUMPSTART_MODEL_ARTIFACT_BUCKET_OVERRIDE)
+
+    if bucket_override:
+        logger.info("Using JumpStart bucket override for model URI: '%s'", bucket_override)
+
     if model_scope == JumpStartScriptScope.INFERENCE:
-        model_artifact_key = (
-            getattr(model_specs, "hosting_prepacked_artifact_key", None)
-            or model_specs.hosting_artifact_key
-        )
+
+        model_artifact_key = None
+
+        for model_artifact_field_name in _MODEL_ARTIFACT_PRIORITY_LIST:
```

Comment:
just from a code legibility standpoint this block would benefit from being extracted into a helper function